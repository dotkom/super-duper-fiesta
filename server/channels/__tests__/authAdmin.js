jest.mock('../../models/meeting');
jest.mock('../../models/user');
jest.mock('../../utils');
const { emit, broadcast } = require('../../utils');
const { createGenfors } = require('../../models/meeting');
const { updateUserById } = require('../../models/user');
const { generateGenfors, generateSocket, generateUser } = require('../../utils/generateTestData');
const { adminLogin, createGenfors: createGenforsListener } = require('../admin/authAdmin');


const MOCK_PW = 'correct';

const generateData = data => Object.assign({}, data);

beforeEach(() => {
  process.env.SDF_GENFORS_ADMIN_PASSWORD = MOCK_PW;
});

describe('admin login', () => {
  it('returns invalid admin password if incorrect', async () => {
    await adminLogin(generateSocket(), generateData());

    expect(emit.mock.calls).toMatchSnapshot();
    expect(broadcast.mock.calls).toEqual([]);
  });

  it('emits sign in with manager permissions if auth successful', async () => {
    updateUserById.mockImplementation(async (id, obj) => Object.assign(generateUser(), obj));
    const user = { permissions: 0 };
    await adminLogin(generateSocket(user),
                     generateData({ password: MOCK_PW }));

    expect(emit.mock.calls).toMatchSnapshot();
    expect(broadcast.mock.calls).toEqual([]);
  });
});

describe('admin creates genfors', () => {
  it('returns invalid admin password if incorrect', async () => {
    await createGenforsListener(generateSocket(), generateData());

    expect(emit.mock.calls).toMatchSnapshot();
    expect(broadcast.mock.calls).toEqual([]);
  });

  it('creates a new meeting if admin password is correct', async () => {
    createGenfors.mockImplementation(async () => generateGenfors());

    await createGenforsListener(generateSocket(), generateData({ password: MOCK_PW }));

    expect(emit.mock.calls).toEqual([]);
    expect(broadcast.mock.calls).toEqual([]);
  });
});

afterEach(() => {
  process.env.SDF_GENFORS_ADMIN_PASSWORD = '';
});
