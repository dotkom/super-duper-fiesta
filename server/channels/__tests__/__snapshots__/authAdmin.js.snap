// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`admin creates genfors returns invalid admin password if incorrect 1`] = `
Array [
  Array [
    "action",
    Object {
      "data": Object {
        "error": "Ugyldig administratorpassord.",
      },
      "type": "ERROR",
    },
  ],
]
`;

exports[`admin login emits sign in with manager permissions if auth successful 1`] = `
Array [
  Array [
    "action",
    Object {
      "data": Object {
        "full_name": "Namy",
        "id": "123",
        "logged_in": true,
        "permissions": 10,
        "username": undefined,
      },
      "type": "AUTH_SIGNED_IN",
    },
  ],
]
`;

exports[`admin login returns invalid admin password if incorrect 1`] = `
Array [
  Array [
    "action",
    Object {
      "data": Object {
        "error": "Ugyldig administratorpassord.",
      },
      "type": "ERROR",
    },
  ],
]
`;
